kokain-processor/src/test/java/com/schwarz/kokain/processor/util/KokainProcessorKotlinTest.kt

```diff
@@ -23,14 +23,28 @@ class KokainProcessorKotlinTest {
 
         val compilation = compileKotlin(subEntity)
 
+        Assert.assertEquals(compilation.exitCode, KotlinCompilation.ExitCode.OK)
+    }
+
+    @Test
+    fun testKotlinInternalBasicGeneration() {
+
+        val subEntity = SourceFile.kotlin("FooBean.kt",
+            HEADER +
+                    "@EBean\n" +
+                    "@EFactory\n" +
+                    "open internal class FooBean {\n" +
+                    "}")
+
+
+        val compilation = compileKotlin(subEntity)
 
-        //TODO Fix classpath problem and make this great again
-        Assert.assertEquals(compilation.exitCode, KotlinCompilation.ExitCode.COMPILATION_ERROR)
+        Assert.assertEquals(compilation.exitCode, KotlinCompilation.ExitCode.OK)
     }
 
     private fun compileKotlin(vararg sourceFiles: SourceFile): KotlinCompilation.Result {
         return KotlinCompilation().apply {
-            sources = sourceFiles.toList()
+            sources = sourceFiles.toMutableList()
```

Comment:
Why does it have to be a mutable list?
I don't see any code adding elements (but I may well have missed it...)